### ISSUE TITLE
[Feature]: Fetch thumbnails with GetCollection

### ISSUE THREAD

**Issue by xmedia-nl** [ToshY/BunnyNet-PHP#111]
### Description

I'd like to fetch the thumbnail of a video I uploaded, and the only way I can find it through the GetCollection endpoint.
If we add a query parameter ?includeThumbnails=true to the path, the response includes the thumbnail links. 

I don't think the current implementation allows adding custom query parameters, so maybe an additional EndpointInterface alongside GetCollection is needed?

### Example

```
class GetCollection implements EndpointInterface
{
    public function getMethod(): Method
    {
        return Method::GET;
    }

    public function getPath(): string
    {
        return 'library/%d/collections/%s**?includeThumbnails=true**';
    }

    public function getHeaders(): array
    {
        return [
            Header::ACCEPT_JSON,
        ];
    }
}
```

**Comment by ToshY** [comment]
Hey @xmedia-nl 👋 

Thanks for creating the issue.

> I'd like to fetch the thumbnail of a video I uploaded, and the only way I can find it through the GetCollection endpoint.
If we add a query parameter ?includeThumbnails=true to the path, the response includes the thumbnail links.

To make sure we are on the same page, it concerns the `Get Collection` from the Stream API: https://docs.bunny.net/reference/collection_getcollection correct? 

In that case, you're right and the option is currently missing; I assume the `includeThumbnails` query parameter was added somewhere in the last couple of months.

> I don't think the current implementation allows adding custom query parameters, so maybe an additional EndpointInterface alongside GetCollection is needed?

Correct, the current `GetCollection` currently does not allow "custom" query parameters. (I guess with "custom" you actually refer to the `includeThumbnails` query parameter)

With that being said, the way I currently setup most classes it makes it relatively easy to add (optional) query parameters to an endpoint by implementing the `getQuery` method:

```php
class GetCollection implements EndpointInterface
{
    public function getMethod(): Method
    {
        return Method::GET;
    }

    public function getPath(): string
    {
        return 'library/%d/collections/%s';
    }

    public function getHeaders(): array
    {
        return [
            Header::ACCEPT_JSON,
        ];
    }

    public function getQuery(): array
    {
        return [
            new AbstractParameter(name: 'includeThumbnails', type: Type::BOOLEAN_TYPE),
        ];
    }
}
```

Checking the API specs, I've noticed that the `Get Collection List` also allows `includeThumbnails` as query parameter, so that means the `ListCollections` class would also need to be updated.

**Comment by xmedia-nl** [comment]
Awesome! That's spot on. Thanks for the swift reply and fix. 
I didn't realize the getQuery() method already existed, as I was only working with the stream API. I tested your fix (bug/111) and it works like a charm

Thanks for your great work! 💪

**Comment by ToshY** [comment]
@xmedia-nl 

This has been released in [4.1.0](https://github.com/ToshY/BunnyNet-PHP/releases/tag/4.1.0).
